devito/types/grid.py

```diff
@@ -305,7 +305,7 @@ def spacing_map(self):
                 # Special case subsampling: `Grid.dimensions` -> (xb, yb, zb)`
                 # where `xb, yb, zb` are ConditionalDimensions whose parents
                 # are SpaceDimensions
-                mapper[d.root.spacing] = s/self.dtype(d.factor.data)
+                mapper[d.root.spacing] = s/self.dtype(d.factor)
```

Comment:
my last worry, after looking at this line, is that this set of changes might somehow break user code, because now `.factor` is a different thing, while the user may have expected a Constant if a Constant was supplied